state/protocol/badger/mutator.go

```diff
@@ -508,16 +517,8 @@ func (m *FollowerState) Finalize(ctx context.Context, blockID flow.Identifier) e
 		return fmt.Errorf("could not retrieve sealed header: %w", err)
 	}
 
-	// THIRD: preparing Epoch-Phase-Change service notifications and metrics updates.
-	// Convention:
-	//                            .. <--- P <----- B
-	//                                    ↑        ↑
-	//             block sealing service event        first block of new
-	//           for epoch-phase transition        Epoch phase (e.g.
-	//              (e.g. EpochSetup event)        (EpochSetup phase)
-	// Per convention, service notifications for Epoch-Phase-Changes are emitted, when
-	// the first block of the new phase (EpochSetup phase) is _finalized_. Meaning
-	// that the new phase has started.
```

Comment:
I feel this old comment https://github.com/onflow/flow-go/blob/4372a12958672917ff8e8b72874a46650163ceb8/state/protocol/badger/mutator.go#L526-L535 is actually quite helpful to retain because we are planning to uniformly wait for finalized seal for the service event

Reply:
It's still there, just modified a bit and moved into the method where the notifications and metrics updates are generated: https://github.com/onflow/flow-go/blob/a562d1f08f0d4c1811887edab1ba309c9230d994/state/protocol/badger/mutator.go#L710-L719